bin/run-tests

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+image_exists() {
+    return $(docker image inspect $1 >/dev/null 2>&1 && true || false)
+}
+
+GIT_SHA=${GITHUB_SHA:-$(git rev-parse HEAD)}
+BUILD="k8s-eip-operator"
+IMAGE="materialize/${BUILD}"
+IMAGE_TAG="${IMAGE}:${GIT_SHA}"
+
+if ! image_exists "$IMAGE_TAG"; then
+    docker build . \
+        -t "$IMAGE_TAG" \
+        -t "$IMAGE:latest"
+fi
```

Comment:
Those are good thoughts -- if we just do `docker build` unconditionally, that should hit the local cache most of the time and not cause too much additional trouble?

What I'd like to do is _not_ tag it as `:latest`, and use the specific tagged image, but I'm not sure how to get that operator image into the KinD cluster for testing without resorting to `kustomize` or doing a `kubectl patch deployment` as part of the test step. Do you have any suggestions for how we could do that more simply?

Reply:
Yeah, I think always running `docker build` is a good solution. If it can't use the cache, it probably needed to be rebuilt anyway.

As for getting the correct tag, ideally we'd just use the image digest instead of a tag, but that might be tricky to get. If we use the shared build image, we need to ensure that the tag is unique per set of code. In CI, the git commit hash should be good, but locally maybe we need to append something like `-dirty` or something.

Reply:
I think I understand what you're saying about `:latest` now. I'm not trying to block the PR for that. If there isn't a nice way to test without a hard-coded tag, we can leave it.

I was more concerned about skipping the build for an image matching that git hash for dirty local builds, which could cause us to test something different than the code we just wrote.

Reply:
@alex-hunt-materialize I just tested this locally and `docker build` did the right thing re: caching so I've just pushed the unconditional change.